tests/components/prometheus/test_init.py

```diff
@@ -250,23 +486,26 @@ async def test_sensor_without_unit(
     """Test prometheus metrics for sensors without a unit."""
     body = await generate_latest_metrics(client)
 
-    assert (
-        'sensor_state{domain="sensor",'
-        'entity="sensor.trend_gradient",'
-        'friendly_name="Trend Gradient"} 0.002' in body
-    )
+    EntityMetric(
+        metric_name="sensor_state",
+        domain="sensor",
+        friendly_name="Trend Gradient",
+        entity="sensor.trend_gradient",
+    ).withValue(0.002).assert_in_metrics(body)
 
-    assert (
-        'sensor_state{domain="sensor",'
-        'entity="sensor.text",'
-        'friendly_name="Text"} 0' not in body
-    )
+    EntityMetric(
+        metric_name="sensor_state",
+        domain="sensor",
+        friendly_name="Text",
+        entity="sensor.text",
+    ).withValue(0.0).assert_not_in_metrics(body)
 
-    assert (
```

Comment:
I think this test was actually broken because the metric is present in the `body` but it is written as `0.0` not `0` so that might be why it fails in upstream? I swapped it in my PR and am open to suggestions around this. I'm not quite sure what we are testing here, tbh

Reply:
I am addressing this in #127262 - would you mind reviewing it please? I don't mind if this PR gets merged first, will be happy to rebase mine.